clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy

```diff
@@ -16,17 +16,45 @@
 
 package com.netflix.spinnaker.clouddriver.openstack.model
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
 import com.google.common.collect.Sets
 import com.netflix.spinnaker.clouddriver.model.LoadBalancer
 import com.netflix.spinnaker.clouddriver.model.LoadBalancerServerGroup
 import com.netflix.spinnaker.clouddriver.openstack.OpenstackCloudProvider
-import groovy.transform.EqualsAndHashCode
+import com.netflix.spinnaker.clouddriver.openstack.domain.LoadBalancerResolver
+import com.netflix.spinnaker.clouddriver.openstack.domain.PoolHealthMonitor
+import groovy.transform.Canonical
+import org.openstack4j.model.network.ext.HealthMonitor
+import org.openstack4j.model.network.ext.LbPool
+
+@Canonical
+@JsonIgnoreProperties(['nameRegex','namePattern','descriptionRegex','descriptionPattern'])
+class OpenstackLoadBalancer implements LoadBalancer, Serializable, LoadBalancerResolver {
+    String type = OpenstackCloudProvider.ID
+    String account
+    String region
+    String id
+    String name
+    String description
+    String status
+    String protocol
+    String method
+    String ip
+    Integer externalPort
+    String subnet
+    Set<PoolHealthMonitor> healthChecks
+    Set<LoadBalancerServerGroup> serverGroups = Sets.newConcurrentHashSet()
+
+  static OpenstackLoadBalancer from(LbPool pool, OpenstackVip vip, OpenstackSubnet subnet, OpenstackFloatingIP ip,
+                                    Set<HealthMonitor> healthMonitors, String account, String region) {
+    new OpenstackLoadBalancer(account: account, region: region, id: pool?.id, name: pool?.name, description: pool?.description,
```

Comment:
It seems weird that `pool` could be null. Would it be better to have that be an error?